File: BE/robocar/src/main/java/com/fiveguys/robocar/repository/UserRepository.java
package com.fiveguys.robocar.repository;

import com.fiveguys.robocar.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserRepository extends JpaRepository<User, Long> {
    boolean existsByLoginId(String loginId);
    boolean existsByNickname(String nickname);
    User deleteByUserId(Long userId);
}

File: AOS/app/src/main/java/org/softeer/robocar/data/repository/user/UserRepository.kt
package org.softeer.robocar.data.repository.user

import org.softeer.robocar.data.dto.user.request.SignUpRequest
import org.softeer.robocar.data.network.ApiResult

interface UserRepository {

    suspend fun checkAvailableUserId(
        userId: String
    ): ApiResult

    suspend fun checkAvailableNickname(
        nickname: String
    ): ApiResult

    suspend fun signUp(
        signUpRequest: SignUpRequest
    ): ApiResult

}